ouf.lua

```diff
@@ -564,6 +564,96 @@ function oUF:Spawn(unit, overrideName)
 	return object
 end
 
+function oUF:SpawnNamePlates(namePrefix, nameplateCallback, nameplateCVars)
+	argcheck(nameplateCallback, 3, 'function', 'nil')
+	argcheck(nameplateCVars, 4, 'table', 'nil')
+
+	local style = style
+	if(not style) then return error('Unable to create frame. No styles have been registered.') end
+
+	local prefix = namePrefix or generateName()
+
+	local hiddenParent = CreateFrame('Frame', nil, UIParent)
+	hiddenParent:SetAllPoints()
+	hiddenParent:Hide()
+
+	NamePlateDriverFrame:Hide()
+	NamePlateDriverFrame:UnregisterAllEvents()
+	NamePlateDriverFrame:RegisterEvent('NAME_PLATE_CREATED')
+	NamePlateDriverFrame:RegisterEvent('NAME_PLATE_UNIT_ADDED')
+	NamePlateDriverFrame:RegisterEvent('NAME_PLATE_UNIT_REMOVED')
+	NamePlateDriverFrame:HookScript('OnEvent', function(_, event, unit)
+		if(event == 'NAME_PLATE_UNIT_ADDED') then
+			local nameplate = C_NamePlate.GetNamePlateForUnit(unit)
+
+			if(nameplate.UnitFrame) then
+				nameplate.UnitFrame:SetParent(hiddenParent)
+			end
+		end
+	end)
+
+	local eventHandler = CreateFrame('Frame')
+	eventHandler:RegisterEvent('NAME_PLATE_UNIT_ADDED')
+	eventHandler:RegisterEvent('NAME_PLATE_UNIT_REMOVED')
+	eventHandler:RegisterEvent('PLAYER_TARGET_CHANGED')
+	eventHandler:RegisterEvent('PLAYER_LOGIN')
+	eventHandler:SetScript('OnEvent', function(_, event, unit)
+		if(event == 'PLAYER_LOGIN') then
+			if(not nameplateCVars) then return end
+
+			for cvar, value in next, nameplateCVars do
```

Comment:
There's a list of 10+ usable CVars, and tables are easy to read/organise, varargs are varargs.